src/data/list/basic.lean

```diff
@@ -3885,4 +3951,45 @@ theorem sum_map_mul_right {α : Type*} [semiring α] {β : Type*} (L : list β)
   (L.map (λ b, f b * r)).sum = (L.map f).sum * r :=
 sum_map_hom L f $ add_monoid_hom.mul_right r
 
+universes u v
+
+@[simp]
+theorem mem_map_swap {α : Type u} {β : Type v} (x : α) (y : β) (xs : list (α × β)) :
+  (y, x) ∈ map prod.swap xs ↔ (x, y) ∈ xs :=
+begin
+  induction xs with x xs,
+  { simp only [not_mem_nil, map_nil] },
+  { cases x with a b,
+    simp only [mem_cons_iff, prod.mk.inj_iff, map, prod.swap_prod_mk, prod.exists, xs_ih],
+    tauto! },
+end
+
+/--
+`list.slice n m xs` removes a slice of length `m` at index `n` in list `xs`.
+-/
+@[simp]
+def slice {α} : ℕ → ℕ → list α → list α
+| 0 n xs := xs.drop n
+| (succ n) m [] := []
+| (succ n) m (x :: xs) := x :: slice n m xs
```

Comment:
* probably belongs in `data/list/defs.lean`?
* could we have lemmas instead of the `@[simp]` attribute, so it's possible to rewrite?
* maybe add the characteristic lemma that this is the same as a combo of `drop` and `take`?